src/test/java/uk/gov/justice/digital/delius/service/AppointmentServiceTest.java

```diff
@@ -32,19 +55,82 @@
     private AppointmentService service;
 
     @BeforeEach
-    @SuppressWarnings("unchecked")
     public void before(){
-        service = new AppointmentService(contactRepository);
-        when(contactRepository.findAll(any(Specification.class), any(Sort.class))).thenReturn(ImmutableList.of());
+        DeliusIntegrationContextConfig integrationContextConfig = new DeliusIntegrationContextConfig();
+        IntegrationContext integrationContext = new IntegrationContext();
+        integrationContextConfig.getIntegrationContexts().put(CONTEXT, integrationContext);
+        integrationContext.setProviderCode(PROVIDER_CODE);
+        integrationContext.setStaffCode(STAFF_CODE);
+        integrationContext.setTeamCode(TEAM_CODE);
+        integrationContext.setRequirementRehabilitationActivityType(RAR_TYPE_CODE);
+        integrationContext.getContactMapping().setAppointmentContactType(CRSAPT_CONTACT_TYPE);
+
+        service = new AppointmentService(contactRepository, requirementService, deliusApiClient, integrationContextConfig);
     }
 
     @Test
     public void appointmentsSortedByContactDateDescending() {
+        when(contactRepository.findAll(any(Specification.class), any(Sort.class))).thenReturn(ImmutableList.of());
         service.appointmentsFor(1L, AppointmentFilter.builder().build());
 
         verify(contactRepository).findAll(specificationArgumentCaptor.capture(), sortArgumentCaptor.capture());
 
         assertThat(sortArgumentCaptor.getValue().getOrderFor("contactDate")).isNotNull();
         assertThat(sortArgumentCaptor.getValue().getOrderFor("contactDate").getDirection()).isEqualTo(Sort.Direction.DESC);
     }
+
+    @Test
+    public void createsAppointment() {
```

Comment:
very minor; but you could use `final var` throughout- which is slightly preferred to cut down on code noise